packages/react-devtools-shared/src/backend/views/TraceUpdates/index.js

```diff
@@ -114,20 +114,22 @@ function prepareToDraw(): void {
   redrawTimeoutID = null;
 
   const now = getCurrentTime();
-  let earliestExpiration = Number.MAX_VALUE;
+  let expirationTimeout = Number.NEGATIVE_INFINITY;
 
   // Remove any items that have already expired.
   nodeToData.forEach((data, node) => {
     if (data.expirationTime < now) {
       nodeToData.delete(node);
     } else {
-      earliestExpiration = Math.min(earliestExpiration, data.expirationTime);
+      expirationTimeout = Math.max(expirationTimeout, data.expirationTime);
```

Comment:
This is wrong though because out of several nodes, it would pick the *later* expiration time. But we want to pick the *earliest* one.